### DISCUSSION TITLE
Parse of std::vector says vector should be reversed?

### DISCUSSION THREAD

**Discussion by JohnBasrai** [CLIUtils/CLI11#951] [Q&A] [answered]
The API docs for void _CLI::App::parse(vector<string&args)_ has the following description

>   The real work is done here. Expects a reversed vector. Changes the vector to the remaining options.

I assume this means that if you want to convert argc/argv to a _std::vector_ to pass here you need to put them in reverse order?  I want to accept a _std::vector_ because I would like to create a wrapper library to expose a more C++ like interface for parsing.  It seems C++ standard still hasn't specified any C++ language version of the 50 year C api for _main_.
Thx.

**Comment by phlptp** [comment, marked as answer]
Take a look at 
https://github.com/CLIUtils/CLI11/blob/main/tests/app_helper.hpp

This is what is used in most of the tests,  put all the arguments in a vector and reverse them, and call the vector parse method.

**Reply by JohnBasrai** [reply]
Thanks, that's what I ended up doing and it's working fine.
